File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        map<int,int> mp1;
        for(int i=0;i<m;i++){
            mp1[nums1[i]]++;
        }
        for(auto it:nums2){
            mp1[it]++;
        }
        nums1.clear();
        for(auto it:mp1){
            while(it.second--){
                nums1.push_back(it.first);
            }
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int l = m - 1, r = n - 1, d=m + n - 1;
        while (r >= 0) {
            if (l >= 0 && nums1[l] > nums2[r]) {
                nums1[d] = nums1[l];
                l--;
            } else {
                nums1[d] = nums2[r];
                r--;
            }
            d--;
        }        
    }
}